roles/common/tasks/networking.yml

```diff
@@ -79,28 +79,53 @@
 #Each host can belong to only one undercloud_cidr, so there will be only one nw_gw set per host.
 - name: set gateway address for multi subnet
   set_fact: nw_gw={{ item.gateway }}
-  when: (update_routes|default('false')|bool) and ((ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') )))
   with_items: "{{ undercloud_cidr }}"
+  when:
+    - update_routes|default('false')|bool
```

Comment:
`update_routes|default(false)` is simper. Because ansible knows update_routes as bool.
Ansible knows all of `True/False, true/false, 'true'/'false', 'True'/'False', "True"/"False"` as bool